peps/pep-0815.rst

```diff
@@ -0,0 +1,2282 @@
+
+- a ``$schema`` key whose value is the URL corresponding to the schema
+  file supplied in the appendix of this PEP. The URL contains the
+  version of the format, and a new version must be added to the appendix
+  whenever the format changes in the future,
+
+- a ``variants`` object listing exactly one variant - the variant
+  provided by the wheel.
+
+The variant.json file corresponding to the wheel built from the example
+pyproject.toml file for x86-64-v3 would look like:
+
+.. code:: json5
+
+    {
+      // TODO: replace this with appendix URL for the PEP
+      "$schema": "https://variants-schema.wheelnext.dev/v0.0.3.json",
+      "default-priorities": {
+         "feature": {
+            "aarch64": ["version"],
+            "x86_64": ["level"]
+         },
+         "namespace": ["x86_64", "aarch64", "blas_lapack"],
+         "property": {
+            "x86_64": {
+               "level": ["v3", "v2", "v1"]
+            }
+         }
+      },
+      "providers": {
+         "aarch64": {
+            "enable-if": "platform_machine == 'aarch64' or 'arm' in platform_machine",
+            "plugin-api": "provider_variant_aarch64.plugin:AArch64Plugin",
+            "requires": [
+               "provider-variant-aarch64 >=0.0.1; python_version >= '3.9'",
+               "legacy-provider-variant-aarch64 >=0.0.1; python_version < '3.9'"
+            ]
+         },
+         "blas_lapack": {
+            "install-time": false,
+            "requires": ["blas-lapack-variant-provider"]
+         },
+         "x86_64": {
+            "enable-if": "platform_machine == 'x86_64' or platform_machine == 'AMD64'",
+            "plugin-api": "provider_variant_x86_64.plugin:X8664Plugin",
+            "requires": ["provider-variant-x86-64 >=0.0.1"]
+         }
+      },
+      "static-properties": {
+         "blas_lapack": {
+            "provider": ["accelerate", "openblas", "mkl"]
+         },
+      },
+      "variants": {
+         // always a single entry, expressing the variant properties of the wheel
+         "x8664v3_openblas": {
+            "blas_lapack": {
+               "provider": ["openblas"]
+            },
+            "x86_64": {
+               "level": ["v3"]
+            }
+         }
+      }
+    }
+
+
+``{name}-{version}-variants.json``: the index level variant metadata file
+'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+For every package version that includes at least one variant wheel,
+there must exist a corresponding ``{name}-{version}-variants.json``
+file, hosted and served by the package index. The ``{name}`` and
+``{version}`` placeholders correspond to the package name and version,
+normalized according to the same rules as wheel files, as found in the
+:ref:`packaging:wheel-file-name-spec` of Binary Distribution Format
+specification.  The link to this file must be present on all index pages
+where the variant wheels are linked.
```

Comment:
We treat the variants json file like an archive file (not like a `.metadata` file), it's a regular top level enty, including having a hash. I'll file a PR.